### PR TITLE
Add Trino 378 release notes

### PR DESCRIPTION
**Pull request by mosabua**
## Description

Assemble the release notes for the upcoming Trino 378 release.

> Is this change a fix, improvement, new feature, refactoring, or other?

Improvement.

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

Documentation only.

> How would you describe this change to a non-technical end user or system administrator?

Add the release notes for Trino 378 with sections for general, security, connectors and other topics as included in the release. 

## Related issues, pull requests, and links

* Fixes https://github.com/trinodb/trino/issues/11947

## Documentation

( ) No documentation is needed.
(✅ ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(✅ ) No release notes entries required.
( ) Release notes entries required with the following suggested text:

## 14 Apr 2022 ✅ 

* https://github.com/trinodb/trino/commit/897e35dd9ac91150f17acbc0b0c3afc833c64b72 @losipiuk #11932 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/6f361f00110f60a6ad7ee27c3bedaa98bec5f041 @losipiuk #11932 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/c8c6739c5da3897c3e3781153bfb33d2a59d8c78 @losipiuk #11932 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/acdb801b6cd25881aed1f7e69b5a05a625e48301 @sopel39 #11738 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/9757ee7238934bb599be90e2a61f820a1b0cc2a7 @sopel39 #11738 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/fd3cb8757a0f88a879de89b18ad1c77c2e224c85 @sopel39 #11738 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/7afef91d61db62a328d182b3fa361c5d59f2c5c9 @sopel39 #11738 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/ef09c4139c588588b6b6a4fb5d32f4af6a00a106 @sopel39 #11738 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/b1b38284aa981065d400cae4ed38a8f525d48237 @arhimondr #11921 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/eb6801d475a7d171ad47ec3091ab4273b0ef57f8 @arhimondr #11921 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/1acf2bd316385f3277992740a1b5abffeb0da356 @findepi #11925 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/e1ba7ebe637e07beb076537ffeb6aa38371eef8b @findepi #11925 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/39aa8bc31a1feb2f09cc4303145093c14aa85f17 @findepi #11925 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/792309aea0433f5c959685bfab5d3356a8626458 @sopel39 #11902 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/7b6fa761ae2ad4dd8ab991456f728332205870b3 @sopel39 #11902 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/ef75c652411305e50ea7ab706e1c6e1cf2f1c355 @sopel39 #11902 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/7fc02df01a9013e1714a8fc8f49ce7acf392785c @martint #11436 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/8b3f2fdd2e5a60a98cbd25c680ef7e590210476e @ebyhr #11969 ✅ rn ✅ docs

## 15 Apr 2022 ✅ 

* https://github.com/trinodb/trino/commit/087868977cc7729621123bbc445384c9dfcae30f @findepi #11965 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/f080e8c06f8aa85bf17c041139a866f3ec19fc14 @findepi #11880 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/81782ee52a280a92e62ea08eecfa31440e991990 @findepi #11918 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/81ab162d03d90ba9b6b2d8219db173ebd3016c8d @findepi #11918 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/b804cdb9d240838921337b3c48e31f2adf5cf25c @findepi #11918 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/3c2c63e509e7ecb28a7fc73c8ee5e412389b0089 @findepi  #11722 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/2df56a057aedf3a59088212214ac84e7ce759f59 @findepi #11977 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/0a220a325828d38ca6783bacf3bcbab58951c41e @findepi #11851 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/926b1bf37d6f44d0cbb1da5e5f0cdca9b1199e14 @findepi #11972 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/ee43fc8fffb640be1cc39944977fa4b9c88af06c @findepi #11877 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/87515b0e612883eebae5662c0f3301616b59b36d @hashhar #7630 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/b7d725227240a16a8436e44b0f161d777c0a9196 @hashhar #11967 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/393986e676c79061d3665d3595edf1aec926aa2a @sopel39 #11983 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/45f1bb2e34153231e566b2a5c41347806f75f7ef @sopel39 #11956 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/ddb583c41fa6bf0c78975e343fc81fe999520c66 @sopel39 #11979  ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/925f4e394fbb0c9ef9958b4a42f454cf5424f8bb @losipiuk #11940  ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/7525907fa21dc773346f3130920c8170ea206158 @losipiuk #11940  ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/1dfac71fc817ea4f6c4327d7cf8afa6a9fcaefbb @hashhar #10562  ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/ba44e5142324889b5955a0bdb48e7503ec315b6a @hashhar  #10562 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/69638d3f39187388983a0f9b457bc87e49e4e4c8 @hashhar  #10562 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/4a5455515a11b3cdcf4fc2926fd6d0d1f523349f @arhimondr #11963 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/feb3b3d213ea5b098ca8949046b005847ab12816 @dain #11944 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/0e1a4da3e5e219814d9a69f980d6dad0ff603110 @ebyhr #11978 ✅ rn ✅ docs

## 16 Apr 2022 ✅  

* no commits

## 17 Apr 2022 ✅ 

* https://github.com/trinodb/trino/commit/146e82aa800e893e02af448ccfabff55276d640b @ebyhr #11661 ✅ rn ✅ docs

## 18 Apr 2022 ✅  

* https://github.com/trinodb/trino/commit/10b2654eaafd6819855d47def04bcba7bfa237ed @ebyhr #11991 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/6b34a29b4c74275adfceafc86872a808e90bd3c0 @ebyhr #11988 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/c76a737eb9875dd54b7baf56d071ff88e34537c5 @arhimondr #11985 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/513d193a2d77799d93358adfc8b5ca2e92b7b410 @ebyhr #12000 ✅ rn ✅ docs

## 19 Apr 2022 ✅ 

* https://github.com/trinodb/trino/commit/b3334abb647131e2a530dca2309d8c02f1fef6b1 @arhimondr #11996 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/8e34a15343c943226c505d59b1378c198ba03c3b @arhimondr #11996 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/070debdd175b8e0b310298819d022440e5940af5 @hashhar #12003 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/2160c4b62c4cedf8bad876aba862e8847902979e @dain #11997 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/cb562e0b36686682465639dc629b0a2490ceef69 @ebyhr #12038 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/025b22a06ba864b9c37d2e58f69c45902502956e @ebyhr #12027 ✅ rn ✅ docs

## 20 Apr 2022 ❌ 

* https://github.com/trinodb/trino/commit/779c2fc956c9abd1c8c80d0cd104e67d0c97bbc2 @ebyhr #12035 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/2cc182e4d0d477cd82263ec05761b25e24385fb4 @findepi #12022 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/e290db1b342b4c4ad023b4e8c91e7ef7eed0815e @findepi #12006 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/5b4248a0b275bc55abeacf11105c20b335bb2751 @findepi #12042 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/4d4d82f973f8b6403ce090d0e9c99e373c0c387a @findepi #11858 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/6d91b210c3c410a9c2acab8d89f85afd17963c01 @findepi #11858 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/a8b461324487bf068a04becafa3eb0b20cec540f @findepi #11858 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/b0dbd3a41deddc93f1db608dada9e4a6ae2e3db6 @hashhar #12025 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/4675f4e72aa750c416a7821fedd439bd817088fc @sopel39 #12051 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/36ff7628d19163a8bc079d7a0fd9ba1c08853e2f @sopel39 #11984 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/42f2c36b9b6135c4fc59fe34c382738148ee01b7 @sopel39 #11984 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/c0812bb5e716db8da979276526efa3cd8aed0f28 @sopel39 #11946 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/aae73d378dac100ddd93dc5c71933b689fb40225 @findepi #11966 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/8ae501a74c2c47c7ba31e6b4b236c71f1e587bf3 @arhimondr #12048 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/a8cb5039128c67d092ba19520d5f66fbfc7f5bcd @arhimondr #11987 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/583b20e92571c07fb2bffe6143a6148c4e854bef @arhimondr #11987 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/85e6ce29b082595a3dedd80cd985e0a39b43f63b @arhimondr #11987 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/e765180fac0c5c6155500085342284c8f0dbd1a6 @arhimondr #12037 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/5b29193b4bf0156e117094e06f17fe98c64efb44 @arhimondr  #12037 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/b4507ddad6e3b4c1cb472dbafb20e5a20b192cf6 @ebyhr #12060 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/d7e6efc21a7dcc1979dae9d495a21434d087ab1f @ebyhr #11957 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/623c50a0d8717dda5c175dd73e336c96ea90152b @ebyhr #11439  ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/1cf972e2dbc0215e00ea7f240515aa70466b2d2f @electrum #11899 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/9c5c8f804b65ef12fe20a5c8beb5cfad4e55d402 @electrum #11894 ✅ rn ❌  docs
* https://github.com/trinodb/trino/commit/82df8d9d098deef72139656df982ae3b0f44ea7b @electrum #12002 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/226c4129560624da09930db3611ab8455d4f79ec @electrum #11998 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/7571a8fd7d1df47fb29a351c7be77335c27ab931 @electrum #12059 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/660109a685ebe3a389aadc66d0de4ad32030bf9b @electrum #12059 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/5ab8d18917c6d8b44388a978e1e9297bdedcc787 @ebyhr #11924 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/eab12243823a5bc3e5195a3dbcb4b27e6699cd02 @ebyhr  #11924 ✅ rn ✅ docs

## 21 Apr 2022 ❌ 

* https://github.com/trinodb/trino/commit/084c45814ae907c6758a692f5aaba9cb83c9a067 @findepi  #12056 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/be8218353f9b43bb02afa52bb9b9f82c3e264942 @findepi #10810 ✅ rn ❌  docs
* https://github.com/trinodb/trino/commit/92adaedbe4d0bebfd2c61724f9a7d98fc5f450bc @findepi #10810 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/1b7654d14d2b05f195450bca1ba739f6d970fe5f @findepi #10810 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/f794cd9f415eb06d023824819fe2fa4c929daac7 @findepi #12023 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/57dd29381a74a6d903b8f6f5b4aaeef114b64284 @sopel39 #12054 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/f4a256ea3d7bdfc758ff6baeb9ef9d82f414702d @sopel39 #12043 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/a3f59967faa72f143e81ee3abc4469fddcff5f43 @findepi #11976 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/c8965bc0cc237acd882551236f4966bc3c832ee5 @findepi #12071 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/d90b7446e1429e149e05b94468901341d517cb7d @ebyhr #12039 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/4ddffd0435e5688005e55bd13b3723af54e61838 @findepi #11980 ❌ rn ✅ docs
* https://github.com/trinodb/trino/commit/d49100bad3bddb6a42a45c33c2f59ab1c92585a9 @findepi #11973 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/4d7a0945f210dafe4c191346ea4a9fa6364bc97d @findepi #11973 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/9395f012821b183341d164007556f747350b5cf2 @findepi #12077 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/0b823cc56f9dc231399dc5af3612d4ef3fd7d9d2 @sopel39 #12047 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/a85f5f74a880e0b7fc10f8338c9986cccf7f46f5 @arhimondr #12065 ✅ rn ✅ docs

### CODE REVIEW

File: docs/src/main/sphinx/release/release-378.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+## General
+
+* Improve planning performance of queries with large `IN` lists.
+  ({issue}`11902`, {issue}`11918`, {issue}`11956`)
+* Fix reporting of total spilled bytes in JMX metrics. ({issue}`11983`)
+
+## Security
+
+* Require value for the ``internal-communication.shared-secret`` configuration
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-378.md:11]: @dain I moved this to Security instead of General and reworded .. ok?

Comment (mosabua) [reply docs/src/main/sphinx/release/release-378.md:11]: Also .. @martint and @dain .. for many users this is a breaking change. How do we best message this? I am going to mention it in the broadcast on Thursday but beyond that?

File: docs/src/main/sphinx/release/release-378.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
+# Release 378 (20 Apr 2022)
+
+## General
+
+* Add {func}`to_base32` and {func}`from_base32` varbinary functions. ({issue}`11439`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-378.md:5]: "Add ... and ... functions". Remove "varbinary".

File: docs/src/main/sphinx/release/release-378.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+## Security
+
+* Require value for the ``internal-communication.shared-secret`` configuration
+  property when authentication is enabled. ({issue}`11944`)
+
+## CLI
+
+* Allow disabling progress reporting during query executing in the CLI client by
+  specifying `--no-progress` ({issue}`11894`)
+* Avoid adding about 200ms of delay to very short queries. ({issue}`11768`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-378.md:19]: > Reduce latency for very short queries

File: docs/src/main/sphinx/release/release-378.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+## Delta Lake connector
+
+* Improve query planning performance. ({issue}`11858`)
+* Fix reading from `information_schema.columns` when metastore contains views. ({issue}`11946`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-378.md:24]: What's fixed?

File: docs/src/main/sphinx/release/release-378.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  double slashes. ({issue}`11964`)
+* Improve query planning performance. ({issue}`11858`)
+
+## MySQL connector
+
+* Improve query planning performance. ({issue}`11858`)
+
+## Pinot connector
+
+* Support querying tables having non-lowercase names in Pinot. ({issue}`6789`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-378.md:44]: `non-lowercase` -> `upper case`?

Comment (mosabua) [reply docs/src/main/sphinx/release/release-378.md:44]: also mixed case and such...  so I think it has to stay as is

**Pull request merged**